### PR TITLE
Legion core / aheal now only removes temp moodies. also bad moodie for using it

### PR DESCRIPTION
**Pull request by ghost**
:cl: Floyd / Qustinnus
fix: Legion core / aheal now only removes temp moodies
add: Legion cores now give you a bad moodlet
/:cl:

#40110

fixes

### CODE REVIEW

**Comment by Qustinnus** [comment]
@Barhandar fix ur thumb thx

**Comment by MimicFaux** [comment]
Idea: refreshes all temp moods, and finishes with a single application of a one bad moodlet for the disgusting squish of a condensed human(?) life force.

**Comment by Ratvar** [comment]
Make negative moodlet permanent and stackable

**Comment by Qustinnus** [comment]
@kevinz000 reason for downvote?

**Comment by ExcessiveUseOfCobblestone** [comment]
Can you edit the title to include the bad moodlet?

File: code/datums/mood_events/generic_negative_events.dm [hunk 1]
```diff
@@ -113,6 +113,11 @@
 	description = "<span class='warning'>I'm missing my family heirloom...</span>\n"
 	mood_change = -4
 
+/datum/mood_event/healsbadman
```
Comment (bgobandit) [inline code/datums/mood_events/generic_negative_events.dm]: honk

File: code/datums/mood_events/generic_negative_events.dm [hunk 2]
```diff
@@ -113,6 +113,11 @@
...
+	description = "<span class='nicegreen'>I feel a lot healthier, but also like shit.</span>\n"
```
Comment (PKPenguin321) [inline code/datums/mood_events/generic_negative_events.dm]: wtf? doesn't it say it feels strange but good or something

Comment (Qustinnus) [reply code/datums/mood_events/generic_negative_events.dm]: wtf why is everyone flipping around what it says. someone told me the complete opposite before

Comment (PKPenguin321) [reply code/datums/mood_events/generic_negative_events.dm]: i might be wrong

Comment (PKPenguin321) [reply code/datums/mood_events/generic_negative_events.dm]: `to_chat(user, "<span class='notice'>You start to smear [src] on yourself. It feels and smells disgusting, but you feel amazingly refreshed in mere moments.</span>")`

>you feel amazingly refreshed

Comment (Qustinnus) [reply code/datums/mood_events/generic_negative_events.dm]: >feels disgusting >shit is disgusting >feels like shit

autism 1 - pkp 0

Comment (PKPenguin321) [reply code/datums/mood_events/generic_negative_events.dm]: 1. bad smell/feel
2. effect kicks in milliseconds later
3. amazingly refreshed feel with no smell

**Comment by Qustinnus** [comment]
I'll show you a magic trick in a bit

File: code/datums/mood_events/generic_negative_events.dm (hunk 2 above)
Comment (Barhandar) [reply code/datums/mood_events/generic_negative_events.dm]: It feels terrible to smear that on you, but your body feels refreshed. Unless it erases your memories you're still going to remember it feeling awful to smear about even though you have no pains, aches or tiredness anymore.

Comment (ExcessiveUseOfCobblestone) [reply code/datums/mood_events/generic_negative_events.dm]: crazy idea: just change the desc to not cause a semantics fit

Comment (Qustinnus) [reply code/datums/mood_events/generic_negative_events.dm]: smh spoiled my magic trick

File: code/datums/components/mood.dm [hunk 3]
```diff
@@ -212,6 +213,16 @@
 	qdel(event)
 	update_mood()
 
+/datum/component/mood/proc/remove_temp_moods(var/admin) //Removes all temp moods
+	for(var/i in mood_events)
+		var/datum/mood_event/moodlet = mood_events[i]
+		if(!moodlet || !moodlet.timeout)
+			return 0
```
Comment (SpaceManiac) [inline code/datums/components/mood.dm]: Seems like this will not remove temporary moodlets which are after a permanent moodlet in the list

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/datums/components/mood.dm (hunk 3 above)
Comment (Qustinnus) [reply code/datums/components/mood.dm]: wow yeah I'm dumb, thanks for pointing that one out.

**Pull request merged**